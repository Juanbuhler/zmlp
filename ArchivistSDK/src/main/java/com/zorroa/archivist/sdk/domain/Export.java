package com.zorroa.archivist.sdk.domain;

/**
 * Created by chambers on 11/1/15.
 */
public class Export {

    private int id;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}
