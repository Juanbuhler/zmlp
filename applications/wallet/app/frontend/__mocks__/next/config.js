import dotenv from 'dotenv'

dotenv.config()

const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
