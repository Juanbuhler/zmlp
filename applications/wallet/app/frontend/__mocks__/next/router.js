/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
