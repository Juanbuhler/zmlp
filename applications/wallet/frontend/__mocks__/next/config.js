const getConfig = () => ({
  serverRuntimeConfig: process.env,
  publicRuntimeConfig: process.env,
})

export default getConfig
