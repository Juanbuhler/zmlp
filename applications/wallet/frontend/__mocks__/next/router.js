/**
 * useRouter
 */

let mockUseRouter = {
  query: {
    action: '',
  },
}

export const __setUseRouter = data => {
  mockUseRouter = data
}

export const useRouter = () => mockUseRouter

/**
 * push
 */

let mockPushFunction = () => {}

export const __setMockPushFunction = fn => {
  mockPushFunction = fn
}

/**
 * on
 */

let mockOnFunction = () => {}

export const __setMockOnFunction = fn => {
  mockOnFunction = fn
}

/**
 * off
 */

let mockOffFunction = () => {}

export const __setMockOffFunction = fn => {
  mockOffFunction = fn
}

/**
 * Router
 */

const Router = {
  push: (...args) => {
    mockPushFunction(...args)
  },
  events: {
    on: (...args) => {
      mockOnFunction(...args)
    },
    off: (...args) => {
      mockOffFunction(...args)
    },
  },
}

export default Router
