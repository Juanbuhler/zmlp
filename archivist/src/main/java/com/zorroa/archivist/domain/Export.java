package com.zorroa.archivist.domain;

/**
 * Created by chambers on 7/11/16.
 */
public class Export {
}
