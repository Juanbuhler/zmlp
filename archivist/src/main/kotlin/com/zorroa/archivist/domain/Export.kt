package com.zorroa.archivist.domain

import com.zorroa.archivist.search.AssetSearch
import java.util.*


/**
 * Defines fields needed to make new ExportFile.
 * @property storageId: The id of the exported file.
 * @property filename: The filename the server will set when downloaded.
 */
data class ExportFileSpec (
        var storageId: String,
        var filename: String
)

/**
 * An ExportFile record.
 */
data class ExportFile (
        val id: UUID,
        val exportId: UUID,
        val name: String,
        val path: String,
        val mimeType : String,
        val size : Long,
        val timeCreated: Long)

/**
 * Defines fields needed to create a new export.
 */
data class ExportSpec (
        var name: String?,
        var search: AssetSearch,
        var processors: List<ProcessorRef> = mutableListOf(),
        var args: Map<String,Any> = mutableMapOf(),
        var env: Map<String,String> = mutableMapOf())


