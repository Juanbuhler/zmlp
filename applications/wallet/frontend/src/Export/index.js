import { useRouter } from 'next/router'

import { spacing } from '../Styles'

import Button, { VARIANTS } from '../Button'

const Export = () => {
  const {
    query: { projectId, query = 'W10=' },
  } = useRouter()

  return (
    <div css={{ padding: spacing.normal }}>
      <div>
        Export the metadata for all assets in the current search as a CSV file. File
        will download automatically.
      </div>

      <div css={{ height: spacing.normal }} />

      <Button
        variant={VARIANTS.PRIMARY_SMALL}
        href={`/api/v1/projects/${projectId}/searches/export/?query=${query}&format=csv`}
        style={{ width: 'fit-content' }}
      >
        Export CSV
      </Button>
    </div>
  )
}

export default Export
