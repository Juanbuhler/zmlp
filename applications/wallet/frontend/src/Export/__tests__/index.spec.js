import TestRenderer from 'react-test-renderer'

import Export from '..'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'

describe('<Export />', () => {
  it('should render properly', () => {
    require('next/router').__setUseRouter({
      query: { projectId: PROJECT_ID },
    })

    const component = TestRenderer.create(<Export />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
