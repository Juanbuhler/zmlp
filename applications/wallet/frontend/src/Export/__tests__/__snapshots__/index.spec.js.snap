// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Export /> should render properly 1`] = `
.emotion-2 {
  padding: 16px;
}

.emotion-0 {
  height: 16px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-1,
.emotion-1:hover,
.emotion-1:visited {
  background-color: #73b61c;
}

.emotion-1:hover {
  background-color: #96c955;
}

.emotion-1[aria-disabled=true] {
  background-color: #808080;
}

<div
  className="emotion-2"
>
  <div>
    Export the metadata for all assets in the current search as a CSV file. File will download automatically.
  </div>
  <div
    className="emotion-0"
  />
  <a
    className="emotion-1"
    href="/api/v1/projects/76917058-b147-4556-987a-0a0f11e46d9b/searches/export/?query=W10=&format=csv"
  >
    Export CSV
  </a>
</div>
`;
